backend/src/routes/notes.ts

```diff
@@ -5,6 +5,12 @@ const router = express.Router();
 
 router.get("/", NotesController.getNotes);
 
+router.get("/trash", NotesController.getDeletedNotes);
+
+router.patch("/trash/:noteId", NotesController.restoreDeletedNote);
```

Comment:
Nit: This route doesn't seem like restoring, I'd do `/trash/restore/:noteId` if we're gonna respect to REST but no change is required here